.ci/benchmarks.groovy

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env groovy
+@Library('apm@current') _
+
+pipeline {
+  agent { label 'linux && immutable' }
+  environment {
+    REPO = 'apm-server'
+    BASE_DIR = "src/github.com/elastic/${env.REPO}"
+    AWS_ACCOUNT_SECRET = 'secret/observability-team/ci/elastic-observability-aws-account-auth'
+    GCP_ACCOUNT_SECRET = 'secret/observability-team/ci/elastic-observability-account-auth'
+    EC_KEY_SECRET = 'secret/observability-team/ci/elastic-cloud/observability-team'
+    TERRAFORM_VERSION = '1.1.9'
+  }
+
+  options {
+    timeout(time: 2, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '100', artifactNumToKeepStr: '30', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+    rateLimitBuilds(throttle: [count: 60, durationName: 'hour', userBoost: true])
+    quietPeriod(10)
+  }
+
+  stages {
+    stage('Checkout') {
+      options { skipDefaultCheckout() }
+      steps {        
+        deleteDir()
+        gitCheckout(basedir: "${BASE_DIR}", shallow: false)        
+      }
+    }
+
+    stage('Benchmarks') {
+      options { skipDefaultCheckout() }
+      steps {
+        dir("${BASE_DIR}/testing/benchmark") {
+          sh(label: 'Build apmbench', script: 'apmbench')
+          withTestClusterEnv {
+            withECKey {
```

Comment:
I've just found [withSecretVault](https://github.com/elastic/apm-pipeline-library/blob/main/vars/withSecretVault.groovy), the below snippet might help to read any given field in a Vault Secret and create an env variable with the value:

```suggestion
            withSecretVault(secret: "${EC_KEY_SECRET}", user_var_name: 'apiKey', pass_var_name: 'EC_API_KEY'){
```


Therefore, the `withECKey` function could be removed.

Reply:
But wait a minute, as I'm working on refactoring `[withSecretVault](https://github.com/elastic/apm-pipeline-library/blob/main/vars/withSecretVault.groovy)` so it can support a more generic data structure

Reply:
https://github.com/elastic/apm-pipeline-library/pull/1704 is the proposal to use something like:

```groovy
withSecretVault(secret: "${EC_KEY_SECRET}", data: ['apiKey': 'EC_API_KEY']){
```

Reply:
if you approve it then I can merge it :)